src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,56 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification. 
+
+## Usage
+
+In order to use the type checker, mount a configuration specification along with this
+plugin. 
+
+`kdb mount <specification> spec/<path> <storage plugin to read the specification> typechecker`
+
+Currently there is no way to check the specification automatically upon mounting. Simply 
+retrieve a random key from the specification using `kdb get <path>` will cause
+the typechecking to happen, issuing a warning if it detects any problem with it.
+
+## Examples
+
+To see its functionality there will be a test specification installed into 
+`/usr/local/share/elektra/specifications/simpleTestSpecification.ini`. This
+path may vary slightly depending on the build configuration. The file can be
+typechecked by mounting it along with the typechecker plugin by calling
+
+`kdb mount simpleTestSpecification.ini spec/examples ini typechecker`
+
+## Debugging
+
+This test specification has no errors by default and will thus report nothing,
+but if you alter it you can experiment with the typechecker. If Elektra is compiled
```

Comment:
Please give some `kdb set` that trigger errors.

Reply:
actually errors may also happen upon a kdb setmeta. Is it possible that in the case of setmeta the plugin won't be activated? I think there is no possibility to run a plugin after a setmeta is there?

Reply:
setmeta and set are the same here, the problem was that neither was present in the original description.